clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessorTest.java

```diff
@@ -231,6 +231,130 @@ public void testSubscriptionChangeEvent() {
         verify(membershipManager, never()).onConsumerPoll();
     }
 
+    @Test
+    public void testSyncCommitEventWithOffsets() {
+        final long deadlineMs = 12345;
+        TopicPartition tp = new TopicPartition("topic", 0);
+        Map<TopicPartition, OffsetAndMetadata> offsets = Collections.singletonMap(tp, new OffsetAndMetadata(10, Optional.of(1), null));
+        SyncCommitEvent event = new SyncCommitEvent(Optional.of(offsets), deadlineMs);
+
+        setupProcessor(true);
+        doReturn(true).when(metadata).updateLastSeenEpochIfNewer(tp, 1);
+        doReturn(CompletableFuture.completedFuture(null)).when(commitRequestManager).commitSync(offsets, deadlineMs);
```

Comment:
Sorry to be daft, but can you explain why we'd want to complete the `Future` with `null` instead of result? I'm confused how the call to `event.future.get()` ends up with the correct value 🤔

Reply:
The future from `CommitRequestManager#commitSync` is `CompletableFuture<Void>`, so it's ok to use `null`. 

The `event.future.get()` can get expected result is because it gets `offsets` here https://github.com/apache/kafka/pull/17150/files#diff-583aa43613edc803471d93ed124d1cbed165768ed674c747784ab068e0ed7518R233.